nx-X11/programs/Xserver/hw/nxagent/Keyboard.c

```diff
@@ -1858,16 +1858,17 @@ void nxagentKeycodeConversionSetup(void)
     char *sessionpath = nxagentGetSessionPath();
     if (sessionpath != NULL)
     {
-      int keyboard_file_path_size = strlen(sessionpath) + strlen("/keyboard");
-      char *keyboard_file_path = malloc((keyboard_file_path_size + 1) * sizeof(char));
+      char *keyboard_file_path = NULL;
       FILE *keyboard_file;
-      if (keyboard_file_path == NULL)
+
+      if ((asprintf(&keyboard_file_path, "%s/keyboard", sessionpath) == -1))
       {
-        FatalError("nxagentKeycodeConversionSetup: malloc failed.");
+        free(sessionpath);
```

Comment:
Okay, but `FatalErrror` is `noreturn` anyway. At this point a memory leak won't hurt. Still good to free the memory.